tests/platform/test_platform.py

```diff
@@ -655,6 +655,33 @@ def test_execute_qprogram_with_qblox(self, platform: Platform):
         # assure only one debug was called
         assert patched_open.call_count == 1
 
+    def test_execute_qprogram_with_qblox_distortions(self, platform: Platform):
+        drive_wf = Square(amplitude=1.0, duration=4)
+        qprogram = QProgram()
+        qprogram.play(bus="drive_line_q0_bus", waveform=drive_wf)
+        qprogram.play(bus="drive_line_q1_bus", waveform=drive_wf)
+
+        test_waveforms_q0 = Waveforms()
+        test_waveforms_q0.add(array=[0.5, 1.0, 0.5, 0.0], index=0)
+        test_waveforms_q0.add(array=[0.0, 0.0, 0.0, 0.0], index=1)
+
+        test_waveforms_q1 = Waveforms()
+        test_waveforms_q1.add(array=drive_wf.envelope(), index=0)
+        test_waveforms_q1.add(array=[0.0, 0.0, 0.0, 0.0], index=1)
+
+        qblox_compiler = QbloxCompiler()
+        sequences, _ = qblox_compiler.compile(qprogram=qprogram)
+        buses = {bus_alias: platform._get_bus_by_alias(alias=bus_alias) for bus_alias in sequences}
+        for bus_alias, bus in buses.items():
+            if bus.distortions:
+                for distrortion in bus.distortions:
+                    for waveforms in sequences[bus_alias]._waveforms._waveforms:  # pylint: disable=protected-access
+                        sequences[bus_alias]._waveforms.modify(  # pylint: disable=protected-access
+                            waveforms.name, distrortion.apply(waveforms.data)
+                        )
+        assert test_waveforms_q0.to_dict() == sequences["drive_line_q0_bus"]._waveforms.to_dict()
+        assert test_waveforms_q1.to_dict() == sequences["drive_line_q1_bus"]._waveforms.to_dict()
+
```

Comment:
You don't call `platform.execute_qprogram()` method at any point. We should always call the method-under-test in our tests, and not replicate its contents, because if the method changes the test will continue to pass.

Check the `test_execute_qprogram_with_qblox()` and `test_execute_qprogram_quantum_machines()`. I can also help you with implementation if you get stuck since these `execute` methods are a bit complex.

Reply:
You can also add the asserts in the `test_execute_qprogram_with_qblox()` directly.

Reply:
I implemented the changes in a different function than `test_execute_qprogram_with_qblox()` with just a play